core.R

```diff
@@ -10,28 +10,61 @@ prepare_dropdown_button <- function(refs_to_list = paste(
                                       "^latest-tag$",
                                       "^v([0-9]+\\.)?([0-9]+\\.)?([0-9]+)$",
                                       sep = "|"
-                                    ), version_tab="") {
+                                    ),
+                                    refs_order = c(
+                                      "devel",
+                                      "pre-release",
+                                      "main",
+                                      "latest-tag"
+                                    ),
+                                    version_tab = "") {
 
-  conf <- eval(parse(text=version_tab))
+  conf <- eval(parse(text = version_tab))
 
   # List and sort versions
   versions <- sort(list.dirs(
     path = ".",
     recursive = FALSE,
     full.names = FALSE
   ), decreasing = TRUE)
+
+  # Filter versions according to refs_to_list
   versions <- versions[grep(refs_to_list, versions)]
+  output <- c()
+
+  # Append non-semantic versions to output vector according to
+  # order in refs_order
+  for (ref in refs_order) {
+    result <- versions[grep(ref, versions)]
+    if (!identical(result, character(0))) {
+      output <- c(output, result)
+    }
+  }
+  semantic_versions <- versions[grepl(
+    "^v([0-9]+\\.)?([0-9]+\\.)?([0-9]+)$",
+    versions
+  )]
```

Comment:
A user could very well not follow semantic versioning for their package. We have that for NEST - where we create tags that are not related to a semantic version.
The better logic here is to `!grepl(preferred order, available versions)`

Reply:
Ok, I corrected that. However, the action will only list versions matching this [regex](https://github.com/insightsengineering/r-pkgdown-multiversion/blob/5d090eac76f8d100c61681daecf12aa91aaf7942/action.yml#L27).

And r.pkg.template doesn't currently have the ability to override it.